flexget/plugins/sites/newznab.py

```diff
@@ -1,124 +1,561 @@
+        if not isinstance(config['search_type'], basestring) and config['search_type'].get('custom_query'):
+            if entry.get('series_name') or entry.get('tvdb_id'):
+                do_show_lookup = True
+            else:
+                do_movie_lookup = True
+
+        # update entry metadata based on task config
+        if config['search_type'] == 'tv' or do_show_lookup:
+            if task.config.get('trakt_lookup'):
+                plugin.get_plugin_by_name('trakt_lookup').instance.lazy_series_lookup(entry)
+                log.verbose('Doing trakt_lookup for < %s >' % entry.get('series_name', entry['title']))
+            if task.config.get('tvmaze_lookup'):
+                plugin.get_plugin_by_name('tvmaze_lookup').instance.lazy_series_lookup(entry)
+                log.verbose('Doing tvmaze_lookup for < %s >' % entry.get('series_name', entry['title']))
+            if task.config.get('thetvdb_lookup') or not entry.get('tvdb_id'):  # default fallback
+                plugin.get_plugin_by_name('thetvdb_lookup').instance.lazy_series_lookup(entry, 'en')
+                log.verbose('Doing thetvdb_lookup for < %s >' % entry.get('series_name', entry['title']))
+        elif config['search_type'] == 'movie' or do_movie_lookup:
+            if task.config.get('trakt_lookup'):
+                plugin.get_plugin_by_name('trakt_lookup').instance.lazy_movie_lookup(entry)
+                log.verbose('Doing trakt_lookup for < %s >' % entry.get('title'))
+            if task.config.get('tmdb_lookup'):
+                plugin.get_plugin_by_name('tmdb_lookup').instance.lookup(entry)
+                log.verbose('Doing tmdb_lookup for < %s >' % entry.get('title'))
+            if task.config.get('imdb_lookup') or not entry.get('imdb_id'):  # default fallback
+                plugin.get_plugin_by_name('imdb_lookup').instance.lookup(entry)
+                log.verbose('Doing imdb_lookup for < %s >' % entry.get('title'))
+
+    def get_metaid_url_parameter(self, task, entry, config):
+        # only use for tv/movie search type
+        if config['search_type'] != 'tv' and config['search_type'] != 'movie':
+            return ''
+
+        self.update_metadata(task, entry, config)
+
+        url_param = ''
+        # use first valid meta id
+        if config['search_type'] == 'tv':
+            if entry.get('tvdb_id'):
+                url_param = '&tvdbid=%s' % entry.get('tvdb_id')
+            elif entry.get('tvrage_id'):
+                url_param = '&rid=%s' % entry.get('tvrage_id')
+            # lets not use those for now, rarely supported!
+            # elif entry.get('tvmaze_series_id'):
+            #    url_param = '&tvmazeid=%s' % entry.get('tvmaze_series_id')
+            # elif entry.get('trakt_id'):
+            #    url_param = '&traktid=%s' % entry.get('trakt_id')
+            if not url_param:
+                log.error('Could not get valid meta id for series < %s >' % (entry.get('series_name', entry['title'])))
```

Comment:
Sure, will change the syntax to match the rest, just my own programmer habit of wanting to distinguish between static and dynamic parts of log values.